### ISSUE TITLE
offline tile source shouldn't crash

### ISSUE THREAD

**Issue by incanus** [mapbox/DEPRECATED-mapbox-ios-sdk#259]
See http://support.mapbox.com/discussions/mapbox-ios-sdk/1143-rmmapboxsource-alloc-initwithmapid-crashes-when-offline for background. This should just not init.

**Comment by RubioV** [comment]
In addition to the crash, the `brandedStringWithContentsOfURL` method contains a blocking call to `[NSURLConnection sendSynchronousRequest...]`. Since the init is usually executed on the main tread, this will block the UI. So even when we're online, this is not good. Apple specifically states:

Important
Because this call can potentially take several minutes to fail (particularly when using a cellular network in iOS), you should never call this function from the main thread of a GUI application.

I don't really know an easy workaround for this, because the result of this function is needed in order to perform a proper init...

**Comment by RubioV** [comment]
Relating to this crash, it is caused by this in `RMConfiguration.m`:

```
    if ( ! returnData)
    {
        *error = internalError;

        return nil;
    }
```

The `error` pointer is not properly checked against `NULL`, so we are dereferencing a `NULL` pointer here. I fixed it by simply removing the `internalError` variable altogether and passing `error` along to sub calls.

After this, there was another crash caused by `RMMapBoxSource`:

```
- (void)dealloc
{
    dispatch_release(_dataQueue);
}
```

`_dataQueue` should first be checked against nil (if it hasn't been properly initialised).

I will submit a pull request for both changes in the coming days.

**Comment by incanus** [comment]
This is addressed in #290. Closing here while I work that out.
